src/gd_interpolation.c

```diff
@@ -1634,7 +1639,7 @@ BGD_DECLARE(gdImagePtr) gdImageScale(const gdImagePtr src, const unsigned int ne
 {
 	gdImagePtr im_scaled = NULL;
 
-	if (src == NULL || src->interpolation_id < 0 || src->interpolation_id > GD_METHOD_COUNT) {
```

Comment:
i don't think this is a good idea.  from what i can tell, the sign of an enum is not guaranteed by the C spec (in fact, it's explicitly left open ended), so deleting this range check opens us up to bugs.

newer C++ standards allow for explicitly controlling the type via scoped enums, but that's only in C++, not C.

we could add a helper func in the header like:
```
static inline bool gdInterpolationMethodIsValid(gdInterpolationMethod id)
{
  return (uintmax_t)id < GD_METHOD_COUNT;
}
```

but that seems like a lot of boiler plate to add a simple cast ...

also, it looks like this code incorrectly lets through GD_METHOD_COUNT when we should return directly.